packages/js/src/utils/watch-for-single-file-changes.ts

```diff
@@ -1,25 +1,26 @@
 import { logger } from '@nrwl/devkit';
-import { join } from 'path';
+import { daemonClient } from 'nx/src/daemon/client/client';
 
 export async function watchForSingleFileChanges(
-  watchDir: string,
+  projectName: string,
   relativeFilePath: string,
   callback: () => void
-) {
-  const watcher = await import('@parcel/watcher');
-  const subscription = await watcher.subscribe(watchDir, (err, events) => {
-    const file = join(watchDir, relativeFilePath);
-    if (err) {
-      logger.error(`Watch error: ${err?.message ?? 'Unknown'}`);
-    } else {
-      for (const event of events) {
-        if (event.path === file) {
-          callback();
-          break;
-        }
+): Promise<() => void> {
+  const unregisterFileWatcher = await daemonClient.registerFileWatcher(
+    { watchProjects: [projectName] },
+    (err, data) => {
+      if (err === 'closed') {
+        logger.error(`Watch error: Daemon closed the connection`);
+        process.exit(1);
+      } else if (err) {
+        logger.error(`Watch error: ${err?.message ?? 'Unknown'}`);
+      } else if (
+        data.changedFiles.some((file) => file.path.includes(relativeFilePath))
```

Comment:
Would this have any false positives? before it was comparing equality, now it is checking substring.

Maybe an edge case, but something like `foo/bar/index.ts` would match `shared/foo/bar/index.ts`.